### PR TITLE
mount2: fixes & updates

### PR DESCRIPTION
**Pull request by darthShadow**
#### What is the purpose of this change?

Various fixes & updates for mount2

#### Was the change discussed in an issue or in the forum before?

N/A

#### Checklist

- [X] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [X] I have added tests for all changes in this PR if appropriate.
- [X] I have added documentation for the changes if appropriate.
- [X] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [X] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

File: cmd/mount2/mount_test.go [hunk 1]
```diff
@@ -1,16 +1,14 @@
-//go:build linux || (darwin && amd64)
-// +build linux darwin,amd64
+//go:build linux
+// +build linux
```
Comment (darthShadow) [inline cmd/mount2/mount_test.go:2]: I limited this test to linux since it was resulting in lockups on the macos runner. Let me know if you want me to revert this change.

Comment (ncw) [reply cmd/mount2/mount_test.go:2]: That is a good idea - thanks.

**Review by ncw — APPROVED** [review]
Looks great - thank you.

Will merge now :-)
